### PR TITLE
simplify constexpr is_hybrid_model<Model> check

### PR DESCRIPTION
**Pull request by Dekken**

### CODE REVIEW

File: src/solver/physical_models/hybrid_model.h [hunk 1]
```diff
@@ -146,17 +143,15 @@ namespace solver
...
     {
-        using type = decltype(hybrid_model_from_type_list(std::declval<TypeList>()));
-    };
+        return std::is_same_v<Model<Args...>, HybridModel<Args...>>;
+    }
 
-    template<typename TypeList>
-    using type_list_to_hybrid_model_t = typename type_list_to_hybrid_model<TypeList>::type;
+    template<typename Model>
+    bool constexpr is_hybrid_model = _is_hybrid_model((Model*)(nullptr));
```
Comment (nicolasaunai) [inline src/solver/physical_models/hybrid_model.h:154]: why Model* ?
how do we know it's going to get a pointer?

Comment (Dekken) [reply src/solver/physical_models/hybrid_model.h:154]: Model* let's us use the type, without declval or decltype.

I don't get your second question.

Comment (Dekken) [reply src/solver/physical_models/hybrid_model.h:154]: we need the type as a function parameter, in order to expand the template from
Type
to
Type<Args...>
which is achieved through parameter type inference 
using a pointer is simple as we don't have an instance of the type, nor do we need one.

Comment (nicolasaunai) [reply src/solver/physical_models/hybrid_model.h:154]: so `Model` is actually `ModelPtr`?

Comment (Dekken) [reply src/solver/physical_models/hybrid_model.h:154]: template<typename Model> <- is not a pointer, and never should be.

if it was, even the current code wouldn't work.

**Pull request closed without merging**